src/Chapter3.hs

```diff
@@ -480,6 +546,81 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
 
+data Person 
+    = Person 
+    deriving (Show)
+
+data House = House 
+    { houseResidents :: [Person]
+    } deriving (Show)
+
+data CityBuilding 
+    = Church 
+    | Library 
+    deriving (Show)
+
+data Castle = Castle
+    { castleName    :: String,
+      cityWalls     :: Int 
+    } deriving (Show)
+
+
+data MagicalCity = MagicalCity
+    { cityBuilding    :: CityBuilding,
+      houses          :: [House],
+      cityCastle      :: Maybe Castle
+    } deriving (Show)
+
+isChurch :: CityBuilding -> Bool
+isChurch (Church) = True
+isChurch _ = False 
+
+hasChurch :: MagicalCity -> Bool
+hasChurch mC = isChurch (cityBuilding mC)
+
+hasCastle :: MagicalCity -> Bool
+hasCastle mC = not (isNothing (cityCastle mC))
+
+wallCount :: Maybe Castle -> Int
+wallCount c
+    | not (isNothing c) = cityWalls (fromJust c)
+    | otherwise = 0
```

Comment:
It is better to pattern match explicitely on the input.
It is not a good practice to use partial functions in your code (`fromJust`, for example), even if you are sure that they are safe there. If there is a way to avoid that it is better to take it 👍🏼